standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -1980,6 +1989,7 @@ public boolean listPartitionsSpecByExpr(String catName, String dbName, String tb
     if (maxParts >= 0) {
       req.setMaxParts(maxParts);
     }
+    req.setValidWriteIdList(validWriteIdList);
```

Comment:
Its confusing to me that we can two different ways to pass down the ValidWriteIdList. I see many other methods in this class which use getValidWriteIdList to set the ValidWriteIdList in the requests. Is it intentional to have them different for some methods? If yes, please document why we do it differently for some methods.

On  a side note, the getValidWriteIdList method retrieves the ValidWriteIdList of the table from conf using the key VALID_TABLES_WRITEIDS_KEY which is set during the lock acquisition phase __after__ query compilation. Doesn't that defeat the purpose of sending the ValidWriteIdList since the objective is to compile the query for a given snapshot?

Reply:
I have added comments in the getValidWriteIdList method in HMS Client.